src/tools/idlc/src/tt_create.c

```diff
@@ -548,6 +567,231 @@ extern void ddsts_struct_empty_close(ddsts_context_t *context, ddsts_type_t **re
   context->cur_type = context->cur_type->type.parent;
 }
 
+extern bool ddsts_add_union_forward(ddsts_context_t *context, ddsts_identifier_t name)
+{
+  assert(cur_scope_is_definition_type(context, DDSTS_MODULE));
+  assert(context->dangling_identifier == name);
+  ddsts_type_t *forward_dcl;
+  dds_return_t rc = ddsts_create_union_forward_dcl(name, &forward_dcl);
+  if (rc != DDS_RETCODE_OK) {
+    context->retcode = rc;
+    return false;
+  }
+  context->dangling_identifier = NULL;
+  ddsts_module_add_member(context->cur_type, forward_dcl);
+  return true;
+}
+
+extern bool ddsts_add_union_open(ddsts_context_t *context, ddsts_identifier_t name)
+{
+  assert(   cur_scope_is_definition_type(context, DDSTS_MODULE)
+         || cur_scope_is_definition_type(context, DDSTS_STRUCT));
+  assert(context->dangling_identifier == name);
+  context->dangling_identifier = NULL;
+  ddsts_type_t *new_union;
+  dds_return_t rc = ddsts_create_union(name, DDSTS_NOTYPE, &new_union);
+  if (rc != DDS_RETCODE_OK) {
+    context->retcode = rc;
+    return false;
+  }
+  context->dangling_identifier = NULL;
+  if (DDSTS_IS_TYPE(context->cur_type, DDSTS_MODULE)) {
+    ddsts_module_add_member(context->cur_type, new_union);
+  }
+  else {
+    ddsts_struct_add_member(context->cur_type, new_union);
+  }
+  context->cur_type = new_union;
+  if (!context_push_annotations_stack(context)) {
+    return false;
+  }
+  return true;
+}
+
+extern bool ddsts_union_set_switch_type(ddsts_context_t *context, ddsts_flags_t base_type)
+{
+  assert(cur_scope_is_definition_type(context, DDSTS_UNION));
+  context->cur_type->union_def.switch_type = base_type;
+  return true;
+}
+
+static const char* name_of_type(ddsts_flags_t flags)
+{
+  switch(flags) {
+    case DDSTS_SHORT:          return "short"; break;
+    case DDSTS_LONG:           return "long"; break;
+    case DDSTS_LONGLONG:       return "long long"; break;
+    case DDSTS_USHORT:         return "unsigned short"; break;
+    case DDSTS_ULONG:          return "unsigned long"; break;
+    case DDSTS_ULONGLONG:      return "unsigned long long"; break;
+    case DDSTS_CHAR:           return "char"; break;
+    case DDSTS_WIDE_CHAR:      return "wide char"; break;
+    case DDSTS_BOOLEAN:        return "boolean"; break;
+    case DDSTS_OCTET:          return "octet"; break;
+    case DDSTS_INT8:           return "int8"; break;
+    case DDSTS_UINT8:          return "uint8"; break;
+    case DDSTS_FLOAT:          return "float"; break;
+    case DDSTS_DOUBLE:         return "double"; break;
+    case DDSTS_LONGDOUBLE:     return "long double"; break;
+    case DDSTS_FIXED_PT_CONST: return "fixed point const"; break;
```

Comment:
Why the `break`s if you have a `return`?

Reply:
Yes, I know it is useless, but I understood that it is good practice to always end a case with a break. It might also prevent some compiler warnings, for compilers that check on a fall-through but do not check if the break statement can be reached (which in this simple case is rather trivial, but not if there a hunderd of lines of code in a case).